### PR TITLE
Benchmark Playground

### PR DESCRIPTION
**Pull request by adi64**
We found this pretty useful for our needs. Instead of settings it up temporarily every time, maybe it makes sense to have a benchmark playground - similar to the regular playground. Like http://quick-bench.com/ but you can use hyrise components.

The added benchmark content is just for illustrational purposes, showcasing a scenario in which it is important to prevent the compiler from optimizing out benchmarked code.

### CODE REVIEW

File: src/benchmark/benchmark_playground.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    });
+  }
+  void TearDown(::benchmark::State& state) override { BenchmarkBasicFixture::TearDown(state); }
+
+ protected:
+  std::vector<ValueT> _vec;
+};
+
+BENCHMARK_F(BenchmarkPlaygroundFixture, BM_Playground_Reference)(benchmark::State& state) {
+  clear_cache();
```
Comment (mrks) [inline src/benchmark/benchmark_playground.cpp]: move to `SetUp`?

File: src/benchmark/benchmark_playground.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  while (state.KeepRunning()) {
+    std::vector<size_t> result;
+    benchmark::DoNotOptimize(result.data());
```
Comment (mrks) [inline src/benchmark/benchmark_playground.cpp]: I would have expected this to be at the end of the benchmark when `result.data()` actually contains content

File: src/benchmark/benchmark_playground.cpp (hunk 1 above)
Comment (adi64) [reply src/benchmark/benchmark_playground.cpp]: done

File: src/benchmark/benchmark_playground.cpp (hunk 2 above)
Comment (adi64) [reply src/benchmark/benchmark_playground.cpp]: Unfortunately, (as far as I understand this correctly), this is not what `benchmark::DoNotOptimize(e)` does. It just marks this memory as globally aliased.
https://stackoverflow.com/questions/40122141/ has a nice explanation.

**Pull request merged**